independent-projects/qute/core/src/main/java/io/quarkus/qute/CompletedStage.java

```diff
@@ -0,0 +1,306 @@
+package io.quarkus.qute;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Internal representation of a completed computation, an optimized replacement for
+ * {@link CompletableFuture#completedFuture(Object)}.
+ * <p>
+ * Note that this is not a full implementation of {@link CompletionStage} - it just throws {@link UnsupportedOperationException}
+ * for methods that are not used internally. Therefore, it should not be used outside the Qute API.
+ */
+public final class CompletedStage<T> implements CompletionStage<T>, Supplier<T> {
```

Comment:
Remove the public statement if it's not be used outside this package?

Reply:
The class must be `public` because it's used by the generated value resolvers...

Reply:
Ah I thought it shouldn't be used or referenced besides here, gotcha